crates/ide-assists/src/utils.rs

```diff
@@ -334,7 +334,7 @@ fn calc_depth(pat: &ast::Pat, depth: usize) -> usize {
 pub(crate) fn find_struct_impl(
```

Comment:
One possible improvement (no need to do this now) would be to have this return/tell which names are already occupied and which aren't. That way we could still show the assist and fill the missing ones.

Reply:
That's a nice idea, I can pick this up in a different PR